packages/app-extensions/src/actions/components/ReportSettings.js

```diff
@@ -12,73 +12,74 @@ import {
 } from '../utils/report'
 import {StyledStickyButtons, StyledReportSettings} from './StyledReportSettings'
 
-export class ReportSettings extends React.Component {
-  constructor(props) {
-    super(props)
-    this.customSettingsDefined
-      = props.settingsDefinition.customSettings && props.settingsDefinition.customSettings.entity
-    this.state = {
-      valid: false,
-      customSettingsValid: !this.customSettingsDefined
-    }
+export const ReportSettings = ({settingsDefinition, formApp, onSubmit, listApp, intl}) => {
+  const customSettingsDefined = settingsDefinition.customSettings && settingsDefinition.customSettings.entity
+  const [settingsChange, setSettingsChange] = useState({
```

Comment:
`customSettingsDefined` is a strange name in my opinion. I would call it just settings.